packages/platforms/react-native/android/src/main/java/com/bugsnag/android/performance/NativeBugsnagPerformanceImpl.java

```diff
@@ -130,9 +140,98 @@ public WritableMap getNativeConfiguration() {
     }
 
     return result;
-  } 
-
+  }
+  
   @Nullable
+  public WritableMap startNativeSpan(String name, ReadableMap options) {
+    if (!isNativePerformanceAvailable) {
+      return null;
+    }
+
+    SpanOptions spanOptions = jsSpanOptionsToNativeSpanOptions(options);
+    SpanFactory spanFactory = BugsnagPerformance.INSTANCE.getInstrumentedAppState$internal().getSpanFactory();
+    SpanImpl nativeSpan = spanFactory.createCustomSpan(name, spanOptions);
+
+    nativeSpan.getAttributes().getEntries$internal().clear();
+
+    WritableMap span = nativeSpanToJsSpan(nativeSpan);
+    return span;
+  }
+
+  private WritableMap nativeSpanToJsSpan(SpanImpl nativeSpan) {
+    WritableMap span = Arguments.createMap();
+    span.putString("name", nativeSpan.getName());
+    span.putString("id", EncodingUtils.toHexString(nativeSpan.getSpanId()));
+    span.putString("traceId", EncodingUtils.toHexString(nativeSpan.getTraceId()));
+
+    long unixNanoStartTime = BugsnagClock.INSTANCE.elapsedNanosToUnixTime(nativeSpan.getStartTime$internal());
+    span.putDouble("startTime", (double)unixNanoStartTime);
+
+    Long parentSpanId = nativeSpan.getParentSpanId();
+    if (parentSpanId > 0L) {
+      span.putString("parentSpanId", EncodingUtils.toHexString(parentSpanId));
+    }
+
+    return span;
+  }
+
+  private SpanOptions jsSpanOptionsToNativeSpanOptions(ReadableMap options) {
+    SpanOptions spanOptions = SpanOptions.DEFAULTS
+      .setFirstClass(true)
+      .makeCurrentContext(false)
+      .within(null);
+
+    if (options.hasKey("startTime")) {
+      double startTime = options.getDouble("startTime");
+      long nativeStartTime = BugsnagClock.INSTANCE.unixNanoTimeToElapsedRealtime((long)startTime);
+      spanOptions = spanOptions.startTime(nativeStartTime);
+    }
+
+    ReadableMap parentContext = null;
+    if (options.hasKey("parentContext") && (parentContext = options.getMap("parentContext")) != null) {
+      SpanContext nativeParentContext = jsSpanContextToNativeSpanContext(parentContext);
+      spanOptions = spanOptions.within(nativeParentContext);
+    }
+
+    return spanOptions;
+  }
+
+  private SpanContext jsSpanContextToNativeSpanContext(ReadableMap spanContext) {
+    String spanId = spanContext.getString("id");
+    String traceId = spanContext.getString("traceId");
+
+    long nativeSpanId = Long.parseUnsignedLong(spanId, 16);
+    UUID nativeTraceId = new UUID(Long.parseUnsignedLong(traceId.substring(0, 16), 16), Long.parseUnsignedLong(traceId.substring(16), 16));
+
+    SpanContext nativeSpanContext = new SpanContext() {
```

Comment:
Any reason this shouldn't maybe be a named class (like `ReactNativeSpanContext`) to remove the noise from this `jsSpanContextToNativeSpanContext`?

Reply:
good idea 👍🏾